public/src/combinationalAnalysis.js

```diff
@@ -296,3 +321,147 @@ function drawCombinationalAnalysis(combinationalData, inputList, outputListNames
+        return;
+    }
+
+    var s = '<table  class="content-table">';
+    s += '<tbody style="display:block; max-height:70vh; overflow-y:scroll" >';
+    s += '<tr>';
+    if ($('#decimalColumnBox').is(':checked')) { s += '<th>' + 'dec' + '</th>'; }
+    for ( i = 0; i < inputListNames.length; i++) { s += `<th>${inputListNames[i]}</th>`; }
+    s += `<th>${booleanExpression}</th>`;
+    s += '</tr>';
+    var matrix = [];
+    for ( i = 0; i < inputListNames.length; i++) {
+        matrix[i] = new Array((inputListNames.length));
+    }
+
+    for ( i = 0; i < inputListNames.length; i++) {
+        for ( j = 0; j < (1 << inputListNames.length); j++) {
+            matrix[i][j] = (+((j & (1 << (inputListNames.length - i - 1))) != 0));
+        }
+    }
+
+    for (i = 0; i < Math.pow(2, inputListNames.length); i++) {
+        let data = [];
+        for (j = 0; j < inputListNames.length; j++) {
+            data[j] = Math.floor(i / Math.pow(2, inputListNames.length - j - 1)) % 2;
+        } 
+        let equation = booleanExpression;
+        for (j = 0; j < inputListNames.length; j++) {
+            equation = equation.replace(new RegExp(inputListNames[j], 'g'), data[j]);
+        }
+
+        output[i] = solve(equation);
+    }
+     
+    for ( j = 0; j < (1 << inputListNames.length); j++) {
+        s += '<tr>';
+        if ($('#decimalColumnBox').is(':checked')) { s += `<td>${j}</td>`; }
+        for ( i = 0; i < inputListNames.length; i++) {
+            s += `<td>${matrix[i][j]}</td>`;
+        }
+        
+        s += `<td class="13" id="${j}">` + `${output[j]}` + '</td>';
+        s += '</tr>';
+    }
+
+    s += '</tbody>';
+    s += '</table>';
+   
+    function solve(equation) {
+        while (equation.indexOf("(") != -1) {
+            let start = equation.lastIndexOf("(");
+            let end = equation.indexOf(")", start);
+            if (start != -1)
+                equation = equation.substring(0, start)
+                    + solve(equation.substring(start + 1, end))
+                    + equation.substring(end + 1);
+        }
+        equation = equation.replace(/''/g, '');
+        equation = equation.replace(/0'/g, '1');
+        equation = equation.replace(/1'/g, '0');
+        for (let i = 0; i < equation.length - 1; i++)
+            if ((equation[i] == '0' || equation[i] == '1') && (equation[i + 1] == '0' || equation[i + 1] == '1'))
+                equation = equation.substring(0, i + 1) + '*' + equation.substring(i + 1, equation.length);
+        try {
+            let safeEval = eval;
+            let answer = safeEval(equation);
+            if (answer == 0)
+                return 0;
+            if (answer > 0)
+                return 1;
+            return '';
+        } catch (e) {
+            return '';
+        }
+    }
+ 
+    $('#combinationalAnalysis').empty();
+    $('#combinationalAnalysis').append(s);
+    $('#combinationalAnalysis').dialog({
```

Comment:
Are you duplicating the code here? Is it possible to reference the old code itself and just update the boolean table there? or am I missing something here.

Reply:
during the process what I have understood is this...(correct me if I misunderstood something!!)

we have to take input in export function ```createcombinationalanalysispromt``` and call functions inside it with arguments as input and scope because through scope a object is generated which is needed to be passed to function which contains timestamps and other autogenerated info according to input variables which will help drawcomboanalysis function to generate circuit

so to treat combinational analysis and boolean function as alternate option we have to use different functions and different dialogues are used for both 

above mentioned code  is inside the function ```createbooleanprompt```  in older code  to use same code for append string and create dialogue box 
we have to call older function from new function  and pass the string and use if else before appending not just here many things will conflict in datapipeline also when we generate circuit 
also combinational analysis have different data  
 so just settled up using different for both
string we are appending is toally different 
the only part duplicating is creating combinational data to generate minimized circuit and truth table print part

guide me if any possibilites ? thanks!!

Reply:
Yeah, so I think you can simply modify the older function as follows: 

`function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope, values = [])`

If values is empty, then set everything to 0, otherwise set it to whatever values u pass. Something along these lines. Dies it make sense?

Reply:
Not sure about this.. but I have a idea 
What we can do is just solve a Boolean expression and return the output in form of array we catch that output array in first function and there only call createbooleanpromt by passing output array and other parameters

As combinational analysis uses 0,x,1 and in Boolean we have to print ouput array we can create a Boolean type  var and pass it to createbooleanpromt and print corresponding truth table using if else 
Remaining part can also be handled using conditional 
I haven't tried it but I will try 
Is this solution ohk with you? @satu0king

Reply:
I didn't follow you but I think I will fix this PR myself, just fix the merge conflict.

Reply:
yup its hard to pass info in words... i wanted to say something presented in screenshot below

![2020-11-20_18-23](https://user-images.githubusercontent.com/55848322/99802390-a4193380-2b5d-11eb-9c9f-0d2cd7be79e3.png)

Reply:
I said exactly this `function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope, values = [])` 
I called it `values`, you called it `output`

Reply:
oops sorry for confusion !!! anyways  so shall i proceed in this way?

Reply:
@pavanjoshi914 can u fix this soon?

Reply:
yes fixing it now thanks!!